dashboard/src/modules/components/OverviewComponent/NewRunsComponent.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState } from "react";
+import "./index.less";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  TableComposable,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Td,
+  InnerScrollContainer,
+  OuterScrollContainer,
+  ActionsColumn,
+  ExpandableRowContent,
+} from "@patternfly/react-table";
+import {
+  deleteDataset,
+  updateDataset,
+  setRows,
+  setSelectedRuns,
+} from "actions/overviewActions";
+import { RenderPagination } from "./common-component";
+
+const NewRunsComponent = () => {
+  const dispatch = useDispatch();
+  const { newRuns, initNewRuns, selectedRuns } = useSelector(
+    (state) => state.overview
+  );
+  const loginDetails = useSelector((state) => state.userAuth.loginDetails);
+
+  const [perPage, setPerPage] = useState(5);
+  const [page, setPage] = useState(1);
+
+  const onSetPage = (_evt, newPage, _perPage, startIdx, endIdx) => {
+    setPage(newPage);
+    dispatch(setRows(newRuns.slice(startIdx, endIdx)));
+  };
+
+  const perPageOptions = [
+    { title: "5", value: 5 },
+    { title: "10", value: 10 },
+    { title: "20", value: 20 },
+  ];
+  const onPerPageSelect = (_evt, newPerPage, newPage, startIdx, endIdx) => {
+    setPerPage(newPerPage);
+    setPage(newPage);
+    dispatch(setRows(newRuns.slice(startIdx, endIdx)));
+  };
+  const columnNames = {
+    result: "Result",
+    endtime: "Endtime",
+  };
+
+  /* Selecting */
+  const areAllRunsSelected =
+    newRuns?.length > 0 && newRuns?.length === selectedRuns?.length;
+  const selectAllRuns = (isSelecting) => {
+    dispatch(setSelectedRuns(isSelecting ? newRuns.map((r) => r) : []));
+  };
+  const onSelectRuns = (repo, __rowIndex, isSelecting) => {
+    const otherSelectedRuns = selectedRuns.filter(
+      (r) => r.resource_id !== repo.resource_id
+    );
+    const c = isSelecting ? [...otherSelectedRuns, repo] : otherSelectedRuns;
+    dispatch(setSelectedRuns(c));
+  };
+  const isRowSelected = (run) =>
+    selectedRuns.filter((item) => item.name === run.name).length > 0;
+  /* Selecting */
+
+  const makeFavorites = (dataset, isFavoriting = true) => {
+    dispatch(updateDataset(dataset, "favorite", isFavoriting));
+  };
+
+  const moreActionItems = (dataset) => [
+    {
+      title: "Save",
+      onClick: () => dispatch(updateDataset(dataset, "save")),
+    },
+    {
+      title: dataset.metadata["dashboard.seen"] ? "Mark unread" : "Mark read",
+      onClick: () =>
+        dispatch(
+          updateDataset(dataset, "read", !dataset.metadata["dashboard.seen"])
+        ),
+    },
+    {
+      title: dataset.metadata["user.favorite"]
+        ? "Mark unfavorite"
+        : "Mark favorite",
+      onClick: () =>
+        dispatch(
+          updateDataset(dataset, "favorite", !dataset.metadata["user.favorite"])
+        ),
+    },
+    {
+      title: "Delete",
+      onClick: () => dispatch(deleteDataset(dataset)),
+    },
+  ];
```

Comment:
Publishing datasets will be allowed only for the saved ones. It will added for the Saved Component

Reply:
Sure, that's fine. I suspect we'll need to get some experience with the workflow here, to decide what makes the most sense for users, but that seems like a reasonable first step.

That is, users might find it awkward that we force them to "save" a dataset before they can choose to "publish" it. Or perhaps that will seem like a completely natural workflow.

Reply:
> users might find it awkward that we force them to "save" a dataset before they can choose to "publish" it

On the flip side, I think it will be awkward if we allow uses to "publish" a dataset which hasn't been "saved", yet.  (Although, I suppose the UI could close that gap by forcing it to be saved when the user tries to publish it.)

Reply:
As a dashboard workflow, that's possible; but remember that "publish" is a Server state that's important to any client, while "saved" is a dashboard-specific concept that doesn't affect server behavior at all.